api/storage_service.cc

```diff
@@ -628,26 +630,25 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
             for (auto& id : table_ids) {
                 co_await db.find_column_family(id).compact_all_sstables();
             }
-            co_return;
-        }).then([]{
-                return make_ready_future<json::json_return_type>(json_void());
-        });
+            });
+
+        co_return json_void();
     });
 
-    ss::force_keyspace_cleanup.set(r, [&ctx, &ss](std::unique_ptr<request> req) {
+    ss::force_keyspace_cleanup.set(r, [&ctx, &ss](std::unique_ptr<request> req) -> future<json::json_return_type> {
         auto keyspace = validate_keyspace(ctx, req->param);
         auto column_families = parse_tables(keyspace, ctx, req->query_parameters, "cf");
         if (column_families.empty()) {
             column_families = map_keys(ctx.db.local().find_keyspace(keyspace).metadata().get()->cf_meta_data());
         }
         apilog.info("force_keyspace_cleanup: keyspace={} tables={}", keyspace, column_families);
-        return ss.local().is_cleanup_allowed(keyspace).then([&ctx, keyspace,
-                column_families = std::move(column_families)] (bool is_cleanup_allowed) mutable {
-            if (!is_cleanup_allowed) {
-                return make_exception_future<json::json_return_type>(
-                        std::runtime_error("Can not perform cleanup operation when topology changes"));
-            }
-            return ctx.db.invoke_on_all([keyspace, column_families] (replica::database& db) -> future<> {
+        if (!co_await ss.local().is_cleanup_allowed(keyspace)) {
+            auto msg = "Can not perform cleanup operation when topology changes";
+            apilog.warn("force_keyspace_cleanup: keyspace={} tables={}: {}", keyspace, column_families, msg);
+            co_await coroutine::return_exception(std::runtime_error(msg));
+        }
+        // FIXME: log error
+            co_await ctx.db.invoke_on_all([&] (replica::database& db) -> future<> {
```

Comment:
Save the captures. `ctx` cannot be accessed here due to `co_await` in 645.

Reply:
Done in v2